src/Model/Endpoint.php

```diff
@@ -213,21 +220,43 @@ public function initialize(array $config)
      *
      * @param string|null $endpoint the new endpoint name
      * @return string
+     * @deprecated 2.0.0 use setName() and getName() instead.
      */
     public function endpoint($endpoint = null)
     {
         if ($endpoint !== null) {
-            $this->_endpoint = $endpoint;
+            return $this->setName($endpoint);
         }
+
+        return $this->getName();
+    }
+
+    /**
+     * Set the name of this endpoint
+     *
+     * @param string $name The name for this endpoint instance
+     * @return $this
+     */
+    public function setName($name)
+    {
+        $inflectMethod = $this->getInflectionMethod();
+        $this->_endpoint = Inflector::$inflectMethod($name);
```

Comment:
It would be better to also rename the property `$_endpoint` to `$_name`. To maintain BC using magic `__get()/__set()` to get / set `$_endpoint`.